.github/workflows/publish.yaml

```diff
@@ -0,0 +1,45 @@
+# .github/workflows/chromatic.yml
+
+# Workflow name
+name: 'Publish Component Library'
+
+# Event for the workflow
+on:
+  pull_request_target:
+    branches:
+      - master
+    types:
+      - closed
+    paths:
+      - packages/components/**            
+  push:
+    branches-ignore:
+      - 'master'
+    paths:
+      - packages/components/**      
+
+# List of jobs
+jobs:
+  npm-publish:
+    # Operating System
+    runs-on: ubuntu-latest
+    # Job steps
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install dependencies, build
+        run: yarn workspace @calyptia-vivo/components install && yarn workspace @calyptia-vivo/components build
+      - name: Configure git
+        run: git config --global user.name "Continuous Integration" && git config --global user.email "ci@calyptia.com"         
+      - name: Publish master
+        if: ${{ github.event_name != 'push' && github.event.pull_request.merged == true }} 
+        run: yarn workspace @calyptia-vivo/components version --patch && yarn workspace @calyptia-vivo/frontend add @calyptia-vivo/components@$npm_package_version
+        env:
+          NPM_AUTH_TOKEN: ${{ secrets.VIVO_NPM_AUTH_TOKEN }}                
+      - name: Publish staging    
+        run: yarn workspace @calyptia-vivo/components version --preid pr --prerelease && yarn workspace @calyptia-vivo/frontend add @calyptia-vivo/components@$npm_package_version
+        env:
+          NPM_AUTH_TOKEN: ${{ secrets.VIVO_NPM_AUTH_TOKEN }}
+        if: ${{ github.event_name == 'push' && github.event.pull_request.merged == false }}
```

Comment:
I think this guard should be below? Not sure what's going on here.

Are you only intending to run this on PRs? Personally it would be easier to split the complex logic here into separate workflows that only trigger on relevant events.